File: docrunner/models/snippet.py
from typing import List

from pydantic import BaseModel

from docrunner.models.snippet_options import SnippetOptions


class Snippet(BaseModel):
    code: str
    options: SnippetOptions

    @classmethod
    def new(cls, code: str, decorators: List[str]):
        return cls(
            code=code, options=SnippetOptions.from_decorators(decorators)
        )

File: lib/models/snippet.dart
import 'snippet_options.dart';

class Snippet {
  Snippet({
    required this.code,
    required this.options,
  });
  String code;
  SnippetOptions options;

  static Snippet create({
    required String code,
    required List<String> decorators,
  }) {
    return Snippet(
      code: code,
      options: SnippetOptions.fromDecorators(decorators),
    );
  }
}
